spec/models/concerns/pageflow/auto_generated_perma_id_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'spec_helper'
+
+module Pageflow
+  describe AutoGeneratedPermaId do
+    describe '#perma_id' do
+      it 'is set on creation' do
+        revision = create(:revision)
+        revision_component = TestRevisionComponent.create!(revision: revision)
+
+        expect(revision_component.perma_id).to be_present
+      end
+
+      it 'differs for separate instances of the same model' do
+        revision = create(:revision)
+        revision_component1 = TestRevisionComponent.create!(revision: revision)
+        revision_component2 = TestRevisionComponent.create!(revision: revision)
+
+        expect(revision_component1.perma_id).not_to eq(revision_component2.perma_id)
+      end
+
+      it 'remains the same on subsequent saves' do
+        revision = create(:revision)
+        revision_component = TestRevisionComponent.create!(revision: revision)
+
+        expect {
+          revision_component.save!
+        }.not_to change { revision_component.perma_id }
+      end
+    end
+  end
+end
```

Comment:
I prefer the way `SerializedConfiguration` is tested below (with an anonymous class). From a super pessimistic point of view, these tests do not ensure that `AutoGeneratedPermaId` actually provides the functionality - only that `RevisionComponent` manages to provide it somehow.